zebra-state/src/service/check.rs

```diff
@@ -100,6 +100,75 @@ where
     Ok(())
 }
 
+/// Check that the `prepared` block is contextually valid for `network`, using
+/// the `history_tree` up to and including the previous block.
+#[tracing::instrument(skip(prepared))]
```

Comment:
You might want to skip `history_tree` here, because its `Debug` impl seems like it could be really long.

(Instrumented fields end up in the logs, so we want them to be around ~20 characters long.)